### PR TITLE
community[patch]: feat: add headers param to cheerio loader

### PR DESCRIPTION
**Pull request by versole**
Feature request: Add a 'fetch headers' parameter to the Cheerio loader to accommodate different results based on the user-agent used.

### CODE REVIEW

File: libs/langchain-community/src/document_loaders/web/cheerio.ts [hunk 1]
```diff
@@ -114,7 +123,8 @@ export class CheerioWebBaseLoader
       this.webPath,
       this.caller,
       this.timeout,
-      this.textDecoder
+      this.textDecoder,
+      this.headers
```
Comment (bracesproul) [inline libs/langchain-community/src/document_loaders/web/cheerio.ts:127]: This is passing headers via the `options` field.

File: libs/langchain-community/src/document_loaders/web/cheerio.ts [hunk 2]
```diff
@@ -92,11 +99,13 @@ export class CheerioWebBaseLoader
     caller: AsyncCaller,
     timeout: number | undefined,
     textDecoder?: TextDecoder,
-    options?: CheerioOptions
+    options?: CheerioOptions,
+    headers?: Headers
```
Comment (bracesproul) [inline libs/langchain-community/src/document_loaders/web/cheerio.ts:103]: Can we combine these? Something like this:

```typescript
static async _scrape(
  ...other args...
  options?: CheerioOptions & { headers?: Headers },
) {
  const { headers, ...cheerioOptions } = options ?? {};
  ...existing code
}
```

**Review by bracesproul — COMMENTED** [review]
overall looks good, left a couple comments. Thanks for adding this!

File: libs/langchain-community/src/document_loaders/web/cheerio.ts (hunk 1 above)
Comment (versole) [reply libs/langchain-community/src/document_loaders/web/cheerio.ts:127]: My bad

File: libs/langchain-community/src/document_loaders/web/cheerio.ts (hunk 2 above)
Comment (versole) [reply libs/langchain-community/src/document_loaders/web/cheerio.ts]: updated

File: libs/langchain-community/src/document_loaders/web/cheerio.ts [hunk 3]
```diff
@@ -78,7 +85,9 @@ export class CheerioWebBaseLoader
     caller: AsyncCaller,
     timeout: number | undefined,
     textDecoder?: TextDecoder,
-    options?: CheerioOptions
+    options?: CheerioOptions & {
+      headers: Headers;
```
Comment (bracesproul) [inline libs/langchain-community/src/document_loaders/web/cheerio.ts:89]: Headers shouldn't be required if options are passed
```suggestion
    options?: CheerioOptions & {
      headers?: Headers;
```

**Review by bracesproul — APPROVED** [review]

File: libs/langchain-community/src/document_loaders/web/cheerio.ts [hunk 4]
```diff
@@ -92,16 +101,20 @@ export class CheerioWebBaseLoader
...
+    }
```
Comment (bracesproul) [inline libs/langchain-community/src/document_loaders/web/cheerio.ts:106]: Same 
```suggestion
    options?: CheerioOptions & {
      headers?: Headers;
    }
```

**Comment by jacoblee93** [comment]
Thank you!

**Pull request merged**